storage/oasis/nodeapi/file/node.go

```diff
@@ -0,0 +1,293 @@
+package file
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"encoding/json"
```

Comment:
Why are we doing both?

In particular, why json? I was surprised by the commit that switches to json. (Thank you for relatively clean commits, btw!)  It adds verbose annotations all over prod types, and it bloats the output by a lot. It's nice to have a human-readable cache, but it's hidden in pogreb's DB anyway. And if wanted to make enable fast reindexing by always writing to cache and essentially reproducing the relevant node state on indexer's disk (that was one of my hopes for this work), we'll end up with hundreds of GB, and then json's verbosity might matter. Though to be honest, it also might not :).

Reply:
Interesting. Thank you for the link. If the first commenter is right, our problem is specifically `Quantity` values that appear in Go maps. Funky.

Regardless, my suggestion here would be to go with cbor everywhere. We know that the types that come from oasis-node can be round-tripped (encode-decode) via cbor, because we already use that functionality. All the custom (de)serializers are implemented and continuously tested in the real world. With JSON and/or gob, I'm slightly concerned that some custom types might change in subtle ways for edge case values, for example, because we never tested all the node types with those encodings.

Reply:
got it; updated to serialize all the data using oasis-core's `cbor`. I kept gob for serializing the cache keys since I'm not sure that oasis-core's `cbor` is meant for this use case.